### PR TITLE
Bug 4686 Fix yahoodatadownloader failing download split and dividend

### PR DESCRIPTION
**Pull request by nazbrok**
#### Description

The decimal parsing in the method YahooDataDownloader.DownloadSplitAndDividendData was throwing and exception when the number was in the exponential format (9.8E-4)

#### Related Issue

related issue : https://github.com/QuantConnect/Lean/issues/4686

#### Motivation and Context

Was not working

#### Requires Documentation Change

No

#### How Has This Been Tested?

Unit test

#### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [x ] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [X] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by Martin-Molinero** [comment]
Hey @nazbrok, Thanks for the fix! `FactorFiles_CanBeGenerated_Accurately` test seems to still be failing with this fix?

**Comment by nazbrok** [comment]
@Martin-Molinero I don't think this is related. The fix was about the conversion issue during the parsing of Dividend values from Yahoo (https://query1.finance.yahoo.com/v7/finance/download/AAPL?period1=0&period2=253402300800&interval=1d&events=dividend&crumb=19LN9kd9jlN).

The test you are refering compare calculated PriceFactor values (indeed from the downloaded yahoo dividend data) to those stored in https://github.com/QuantConnect/Lean/blob/96da867109296a2fc44e78a72f6f908ab4fa791a/Data/equity/usa/factor_files/aapl.csv

The calculated values are different and since this part of code didn't move from ages, as far as I can see, nor the aapl.csv file I wonder if the data returned by Yahoo changed (is that possible ?).

Here is the formula used to calculate the PriceFactor

https://github.com/QuantConnect/Lean/blob/96da867109296a2fc44e78a72f6f908ab4fa791a/ToolBox/FactorFileGenerator.cs#L229-L233

In our case, the first one calculated if for the date 10/05/2018.

The data used from Yahoo are : 

```
Date | Dividends
11/05/18 | 0.1825
```
The previousClosingPrice.Close is 190.03

Since it's the first value calculated, here `previousFactorFileRow ` is initialized like this

https://github.com/QuantConnect/Lean/blob/96da867109296a2fc44e78a72f6f908ab4fa791a/ToolBox/FactorFileGenerator.cs#L76-L80

So according to the PriceFactor formula we have 

`PriceFactor = 1 - 0.1825 / 1 / 190.03 / 1 = **0.9990396**`

While in AAPL.csv we have for this date a PriceFactor of **0.9961585**

image



According to those sites the historical dividend for this date is not 0.1825 like yahoo told us but 0.73
 https://www.nasdaq.com/market-activity/stocks/aapl/dividend-history
https://www.streetinsider.com/dividend_history.php?q=AAPL

image

If I calculate the price factor using $0.73 I get 

`PriceFactor = 1 - 0.73 / 1 / 190.03 / 1 = **0.9961585**`

Which is our calculated pricefactor present in AAPL.csv


I checked on Yahoo's page if the data shown were the same as returned and it is

image
 

So in short, to me Yahoo is not reliable for dividends historical data on AAPL.
I think my PR correctly fix the initial bug but there is another issue to create for the Yahoo dividend historical data for AAPL

**Comment by jaredbroad** [comment]
Great analysis @nazbrok. I 💯 agree yahoo data is not reliable but your fix seems solid for the E-issue. 👍

**Review by Martin-Molinero — APPROVED** [review]
Thanks! 👍
